src/workerd/io/trace-stream.c++

```diff
@@ -826,6 +826,14 @@ class TailStreamTarget final: public rpc::TailStreamTarget::Server {
     // the stream should be stopped and will fulfill the done promise.
     bool finishing = false;
 
+    // When we have a bona fide tail worker that receives its outcome event, we need to make sure
+    // that the final tail worker invocation is completed before destroying the tail worker
+    // customEvent and incomingRequest. To achieve this, we only fulfill the doneFulfiller after
+    // JS execution has completed.
+    // TODO(streaming-tail): I doubt this is the best possible approach here. We should be able to
```

Comment:
Not sure how to handle this well so that the JS call gets to complete before wrapping up the customEvent, but without stalling other actions unnecessarily – "option 2" below solves the problems as far as tests are concerned, but it's certainly inelegant. "option 1" results in some errors but should be cleaner. (We probably need to discuss this out-of-band).

Reply:
Ended up getting option 1 to work – turns out JSG didn't like me returning `jsg::Promise<nullptr>`, removing that fixes error messages during the test. Also added an error handler to reject the fulfiller if needed.